nifi-nar-bundles/nifi-jolt-record-bundle/nifi-jolt-record-processors/src/main/java/org/apache/nifi/processors/jolt/record/JoltTransformRecord.java

```diff
@@ -0,0 +1,522 @@
+
+            } else {
+                transformedObjects = recordList.stream().map((r) -> transform(transform, r)).collect(Collectors.toList());
+            }
+
+            // JOLT expects arrays to be of type List where our Record code uses Object[].
+            // Make another pass of the transformed objects to change List to Object[].
+            if (transformedObjects instanceof Map) {
+                // The set of incoming records has been transformed to a single record (Map)
+                List<Record> normalizedList = new ArrayList<>();
+                normalizedList.add(DataTypeUtils.toRecord(normalizeRecordObjects(transformedObjects), schema, "r"));
+                transformedObjects = normalizedList;
+
+            } else if (transformedObjects instanceof List) {
+                transformedObjects = ((List) transformedObjects).stream()
+                        .map(JoltTransformRecord::normalizeRecordObjects)
+                        .map((o) -> DataTypeUtils.toRecord(o, schema, "r"))
+                        .collect(Collectors.toList());
+            }
+
+            transformedRecordSet = new ListRecordSet(schema, (List<Record>) transformedObjects);
+
+        } catch (final Exception ex) {
+            logger.error("Unable to transform {} due to {}", new Object[]{original, ex.toString(), ex});
+            session.transfer(original, REL_FAILURE);
+            return;
+        } finally {
+            if (customClassLoader != null && originalContextClassLoader != null) {
+                Thread.currentThread().setContextClassLoader(originalContextClassLoader);
+            }
+        }
+
+        FlowFile transformed = session.create(original);
+        final Map<String, String> attributes = new HashMap<>();
+        try (final OutputStream out = session.write(transformed);
+             final RecordSetWriter writer = writerFactory.createWriter(getLogger(), schema, out)) {
+            final WriteResult writeResult = writer.write(transformedRecordSet);
+            attributes.put("record.count", String.valueOf(writeResult.getRecordCount()));
+            attributes.put(CoreAttributes.MIME_TYPE.key(), writer.getMimeType());
+            attributes.putAll(writeResult.getAttributes());
+        } catch (Exception e) {
+            logger.error("Unable to write transformed records {} due to {}", new Object[]{original, e.toString(), e});
+            session.remove(transformed);
+            session.transfer(original, REL_FAILURE);
+            return;
+        }
+
+        final String transformType = context.getProperty(JOLT_TRANSFORM).getValue();
+        transformed = session.putAllAttributes(transformed, attributes);
+        session.transfer(transformed, REL_SUCCESS);
+        session.getProvenanceReporter().modifyContent(transformed, "Modified With " + transformType, stopWatch.getElapsed(TimeUnit.MILLISECONDS));
+        session.remove(original);
+        logger.info("Transformed {}", new Object[]{original});
```

Comment:
Debug level instead?

Reply:
yeah, that's a copy-paste from JoltTransformJSON, should be debug instead